src/Symfony/Component/Console/Tests/Descriptor/AbstractDescriptorTestCase.php

```diff
@@ -53,27 +53,27 @@ public function testDescribeApplication(Application $application, $expectedDescr
 
     public static function getDescribeInputArgumentTestData()
     {
-        return static::getDescriptionTestData(ObjectsProvider::getInputArguments());
+        return self::getDescriptionTestData(ObjectsProvider::getInputArguments());
     }
 
     public static function getDescribeInputOptionTestData()
     {
-        return static::getDescriptionTestData(ObjectsProvider::getInputOptions());
+        return self::getDescriptionTestData(ObjectsProvider::getInputOptions());
     }
 
     public static function getDescribeInputDefinitionTestData()
     {
-        return static::getDescriptionTestData(ObjectsProvider::getInputDefinitions());
+        return self::getDescriptionTestData(ObjectsProvider::getInputDefinitions());
     }
 
     public static function getDescribeCommandTestData()
     {
-        return static::getDescriptionTestData(ObjectsProvider::getCommands());
+        return self::getDescriptionTestData(ObjectsProvider::getCommands());
     }
 
     public static function getDescribeApplicationTestData()
     {
-        return static::getDescriptionTestData(ObjectsProvider::getApplications());
+        return self::getDescriptionTestData(ObjectsProvider::getApplications());
```

Comment:
`getDescriptionTestData` is protected, but switching to `self::` makes it ignore any override of the method. Is it expected ?

Reply:
It isn't. It actually works here because the method is currently not overridden. But it would definitely cause some problems if it's overridden one day. I addressed both your comments in https://github.com/symfony/symfony/pull/49328, thank you very much for letting me know!